test/modules/post/test/cmd_exec.rb

```diff
@@ -111,4 +111,63 @@ def test_cmd_exec_stderr
       end
     end
   end
+
+  def test_cmd_exec_channelized
+    vprint_status("Starting cmd_exec channelized tests")
+
+    it "should run channelized command and wait for output with an extra channel" do
+      test_string = Rex::Text.rand_text_alpha(4)
+      timeout = 15
+      channel_count = session.channels.count
+
+      if session.platform.eql? 'windows'
+        output = cmd_exec("cmd.exe", "/c echo #{test_string}", timeout, { 'Channelized' => true})
+        output = output.rstrip
+      else
+        output = cmd_exec("echo #{test_string}", nil, timeout, { 'Channelized' => true})
+      end
+
+      output == test_string && session.channels.count == channel_count + 1
```

Comment:
Just confirming if this line is valid, I would have thought that the channel count remains the same after a `cmd_exec` exits 🤔 

Are channels closed out of band, or is this an existing bug maybe 👀

Reply:
Correct me if I'm wrong in what I am about to say, but I believe that is an issue on the Metasploit Framework side of things rather than Meterpreter. The channels are not deleted from memory, even though we have received their `core_channel_close` request and they are in theory not interact-able. In theory, your comment is valid and should be the way it operates but currently the channel count just increases.

Reply:
Yip, I believe so :+1: Looks like we'll need to update this then 📈

Reply:
@adfoster-r7 If this needs to be reassigned feel free to take me off this until we are ready to review this again, sounds like the tests here may need to be updated and a new PR put up to fix the core issue at hand here.

Reply:
Sounds good, thanks! Will draft this for a few cycles :+1: